sql/src/main/java/io/crate/planner/operators/LogicalPlanner.java

```diff
@@ -109,7 +110,8 @@
         new MoveOrderBeneathNestedLoop(),
         new MoveOrderBeneathBoundary(),
         new MoveOrderBeneathFetchOrEval(),
-        new DeduplicateOrder()
+        new DeduplicateOrder(),
+        new MoveFilterBeneathHashJoin()
```

Comment:
just curious: any reason to not add it below the other `MoveFilter...` rules but at the bottom instead?

Reply:
No reason - I've changed it to be grouped with the other `MoveFilter...` and also added the same rule for nested loops (non-outer-joins)